pkg/server/sandbox_status.go

```diff
@@ -82,62 +86,70 @@ func (c *criContainerdService) PodSandboxStatus(ctx context.Context, r *runtime.
 
 // TODO (mikebrow): discuss predefining constants structures for some or all of these field names in CRI
 type sandboxInfo struct {
-	Pid         uint32            `json:"pid"`
-	State       string            `json:"state"`
-	SnapshotKey string            `json:"snapshotKey"`
-	Snapshotter string            `json:"snapshotter"`
-	RuntimeSpec *runtimespec.Spec `json:"runtimeSpec"`
+	Pid         uint32                    `json:"pid"`
+	Status      string                    `json:"processStatus"`
+	NetNS       string                    `json:"netNamespace"`
+	NetNSClosed bool                      `json:"netNamespaceClosed"`
+	Image       string                    `json:"image"`
+	SnapshotKey string                    `json:"snapshotKey"`
+	Snapshotter string                    `json:"snapshotter"`
+	Config      *runtime.PodSandboxConfig `json:"config"`
+	RuntimeSpec *runtimespec.Spec         `json:"runtimeSpec"`
 }
 
-// toCRISandboxContainerInfo converts internal container object information to CRI sandbox container status response info map.
-func toCRISandboxContainerInfo(ctx context.Context, container containerd.Container, verbose bool) (map[string]string, error) {
+// toCRISandboxInfo converts internal container object information to CRI sandbox status response info map.
+func toCRISandboxInfo(ctx context.Context, sandbox sandboxstore.Sandbox,
+	pid uint32, processStatus containerd.ProcessStatus, verbose bool) (map[string]string, error) {
 	if !verbose {
 		return nil, nil
 	}
 
-	task, err := container.Task(ctx, nil)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get sandbox container task: %v", err)
-	}
-	status, err := task.Status(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get sandbox container task status: %v", err)
+	si := &sandboxInfo{
+		Pid:    pid,
+		Status: string(processStatus),
+		NetNS:  sandbox.NetNSPath,
+		Config: sandbox.Config,
 	}
 
-	info := make(map[string]string)
-	pid := task.Pid()
```

Comment:
correct pid.

Reply:
For running sandbox, it is non-zero:
```
"info": {
    "pid": 26174,
    "processStatus": "running",
    "netNamespace": "/var/run/netns/cni-343162ed-9064-64c8-2ed8-b6a6f8213e0e",
    "netNamespaceClosed": false,
    "image": "gcr.io/google_containers/pause:3.0",
    "snapshotKey": "d2ec0574ba0f8cf0ed9bc14e5b7cc7dc365567bb2456cc315e0d3547a4bea5e5",
    "snapshotter": "overlayfs",
    "config": {
      "metadata": {
        "name": "nginx-sandbox",
        "uid": "hdishd83djaidwnduwk28bcsb",
        "namespace": "default",
        "attempt": 1
      },
```

Reply:
looks good .. I saw 0 before not sure what changed but good!